### PR TITLE
Refactor admin/security to use Composition API

### PR DESCRIPTION
**Pull request by pixeldesu**
# What

Use Composition API for `admin/security` page and its subcomponents

# Why

Refactor to Composition API (per contribution guide)

### CODE REVIEW

File: packages/client/src/pages/admin/bot-protection.vue [hunk 1]
```diff
@@ -54,64 +54,39 @@ import * as os from '@/os';
...
-	}
-});
+function save() {
+	os.apiWithDialog('admin/update-meta', {
+		enableHcaptcha: enableHcaptcha,
+		hcaptchaSiteKey: hcaptchaSiteKey,
+		hcaptchaSecretKey: hcaptchaSecretKey,
+		enableRecaptcha: enableRecaptcha,
+		recaptchaSiteKey: recaptchaSiteKey,
+		recaptchaSecretKey: recaptchaSecretKey,
```
Comment (Johann150) [inline packages/client/src/pages/admin/bot-protection.vue]:
```suggestion
		enableHcaptcha,
		hcaptchaSiteKey,
		hcaptchaSecretKey,
		enableRecaptcha,
		recaptchaSiteKey,
		recaptchaSecretKey,
```

File: packages/client/src/pages/admin/security.vue [hunk 2]
```diff
@@ -42,49 +42,32 @@ import XBotProtection from './bot-protection.vue';
...
-				bg: 'var(--bg)',
-			},
-			summalyProxy: '',
-			enableHcaptcha: false,
-			enableRecaptcha: false,
-		}
-	},
+function save() {
+	os.apiWithDialog('admin/update-meta', {
+		summalyProxy: summalyProxy,
```
Comment (Johann150) [inline packages/client/src/pages/admin/security.vue]:
```suggestion
		summalyProxy,
```

**Review by Johann150 — APPROVED** [review]
Changes are good, just some minor stylistic improvements.

**Comment by syuilo** [comment]
🙏🏻

**Pull request merged**